ngrep

ngrep is currently failing to build like seen below.
```
yamlfile is ngrep.yaml
Building package ngrep with version ngrep-1.47-r1 from file ngrep.yaml
make yamlfile=ngrep.yaml pkgname=ngrep packages/x86_64/ngrep-1.47-r1.apk
make[1]: Entering directory '/home/user/src/wolfi-os'
@SOURCE_DATE_EPOCH=1715875498 /usr/bin/melange build ngrep.yaml --repository-append /home/user/src/wolfi-os/packages --keyring-append local-melange.rsa.pub --signing-key local-melange.rsa --arch x86_64 --env-file build-x86_64.env --namespace wolfi --generate-index false  --pipeline-dir ./pipelines/  -k https://packages.wolfi.dev/os/wolfi-signing.rsa.pub -r https://packages.wolfi.dev/os
2024/08/29 20:12:33 INFO melange is building:
2024/08/29 20:12:33 INFO   configuration file: ngrep.yaml
2024/08/29 20:12:33 INFO   workspace dir: /home/user/tmp/melange-workspace-2747928236
2024/08/29 20:12:33 INFO evaluating pipelines for package requirements
2024/08/29 20:12:33 INFO   adding package "git" for pipeline "Check out sources from git"
2024/08/29 20:12:33 INFO   adding package "autoconf" for pipeline "Run autoconf configure script"
2024/08/29 20:12:33 INFO   adding package "automake" for pipeline "Run autoconf configure script"
2024/08/29 20:12:33 INFO   adding package "make" for pipeline "Run autoconf make install"
2024/08/29 20:12:33 INFO   adding package "binutils" for pipeline "Strip binaries"
2024/08/29 20:12:33 INFO   adding package "scanelf" for pipeline "Strip binaries"
2024/08/29 20:12:33 INFO --cache-dir ./melange-cache/ not a dir; skipping
2024/08/29 20:12:33 INFO populating workspace /home/user/tmp/melange-workspace-2747928236 from ./ngrep/
2024/08/29 20:12:33 INFO building workspace in '/home/user/tmp/melange-guest-2689419024' with apko
2024/08/29 20:12:33 INFO setting apk repositories: [/home/user/src/wolfi-os/packages https://packages.wolfi.dev/os]
2024/08/29 20:12:33 INFO image configuration:
2024/08/29 20:12:33 INFO   contents:
2024/08/29 20:12:33 INFO     build repositories: []
2024/08/29 20:12:33 INFO     runtime repositories: []
2024/08/29 20:12:33 INFO     keyring:      []
2024/08/29 20:12:33 INFO     packages:     [autoconf automake build-base busybox ca-certificates-bundle libpcap-dev pcre-dev git autoconf automake make binutils scanelf]
2024/08/29 20:12:33 INFO   accounts:
2024/08/29 20:12:33 INFO     runas:  
2024/08/29 20:12:33 INFO     users:
2024/08/29 20:12:33 INFO       - uid=1000(build) gid=1000
2024/08/29 20:12:33 INFO     groups:
2024/08/29 20:12:33 INFO       - gid=1000(build) members=[build]
2024/08/29 20:12:33 INFO auth configured for: []
2024/08/29 20:12:34 INFO installing ca-certificates-bundle (20240705-r0)
2024/08/29 20:12:34 INFO installing wolfi-baselayout (20230201-r15)
2024/08/29 20:12:34 INFO installing ld-linux (2.40-r1)
2024/08/29 20:12:34 INFO installing glibc-locale-posix (2.40-r1)
2024/08/29 20:12:34 INFO installing glibc (2.40-r1)
2024/08/29 20:12:34 INFO installing m4 (1.4.19-r4)
2024/08/29 20:12:34 INFO installing libbz2-1 (1.0.8-r8)
2024/08/29 20:12:34 INFO installing zlib (1.3.1-r4)
2024/08/29 20:12:34 INFO installing libxcrypt (4.4.36-r7)
2024/08/29 20:12:34 INFO installing libcrypt1 (2.40-r1)
2024/08/29 20:12:34 INFO installing perl (5.40.0-r3)
2024/08/29 20:12:34 INFO installing autoconf (2.72-r0)
2024/08/29 20:12:34 INFO installing automake (1.17-r0)
2024/08/29 20:12:34 INFO installing libzstd1 (1.5.6-r4)
2024/08/29 20:12:34 INFO installing libgcc (14.2.0-r2)
2024/08/29 20:12:34 INFO installing libstdc++ (14.2.0-r2)
2024/08/29 20:12:34 INFO installing binutils (2.43.1-r0)
2024/08/29 20:12:34 INFO installing make (4.4.1-r3)
2024/08/29 20:12:34 INFO installing pkgconf (2.3.0-r0)
2024/08/29 20:12:34 INFO installing libgo (14.2.0-r2)
2024/08/29 20:12:34 INFO installing posix-cc-wrappers (1-r3)
2024/08/29 20:12:34 INFO installing libquadmath (14.2.0-r2)
2024/08/29 20:12:34 INFO installing gmp (6.3.0-r2)
2024/08/29 20:12:34 INFO installing isl (0.26-r4)
2024/08/29 20:12:34 INFO installing mpfr (4.2.1-r4)
2024/08/29 20:12:34 INFO installing mpc (1.3.1-r5)
2024/08/29 20:12:34 INFO installing libstdc++-dev (14.2.0-r2)
2024/08/29 20:12:34 INFO installing libatomic (14.2.0-r2)
2024/08/29 20:12:34 INFO installing libgomp (14.2.0-r2)
2024/08/29 20:12:34 INFO installing gcc (14.2.0-r2)
2024/08/29 20:12:34 INFO installing libxcrypt-dev (4.4.36-r7)
2024/08/29 20:12:34 INFO installing linux-headers (6.6.29-r1)
2024/08/29 20:12:34 INFO installing nss-hesiod (2.40-r1)
2024/08/29 20:12:34 INFO installing nss-db (2.40-r1)
2024/08/29 20:12:34 INFO installing glibc-dev (2.40-r1)
2024/08/29 20:12:34 INFO installing build-base (1-r8)
2024/08/29 20:12:34 INFO installing busybox (1.36.1-r10)
2024/08/29 20:12:34 INFO installing libexpat1 (2.6.2-r1)
2024/08/29 20:12:34 INFO installing libpcre2-8-0 (10.44-r1)
2024/08/29 20:12:34 INFO installing libunistring (1.2-r2)
2024/08/29 20:12:34 INFO installing libidn2 (2.3.7-r2)
2024/08/29 20:12:34 INFO installing libpsl (0.21.5-r3)
2024/08/29 20:12:34 INFO installing libbrotlicommon1 (1.1.0-r4)
2024/08/29 20:12:34 INFO installing libbrotlidec1 (1.1.0-r4)
2024/08/29 20:12:34 INFO installing libnghttp2-14 (1.63.0-r0)
2024/08/29 20:12:34 INFO installing libcrypto3 (3.3.1-r5)
2024/08/29 20:12:34 INFO installing libssl3 (3.3.1-r5)
2024/08/29 20:12:34 INFO installing libcurl-openssl4 (8.9.1-r3)
2024/08/29 20:12:34 INFO installing git (2.46.0-r2)
2024/08/29 20:12:34 INFO installing libpcap (1.10.4-r3)
2024/08/29 20:12:34 INFO installing libpcap-dev (1.10.4-r3)
2024/08/29 20:12:34 INFO installing pcre (8.45-r3)
2024/08/29 20:12:34 INFO installing pcre-dev (8.45-r3)
2024/08/29 20:12:34 INFO installing scanelf (1.3.7-r2)
2024/08/29 20:12:36 INFO built image layer tarball as /home/user/tmp/apko-temp-632393022/apko-x86_64.tar.gz
2024/08/29 20:12:36 INFO using /home/user/tmp/apko-temp-632393022/apko-x86_64.tar.gz for image layer
2024/08/29 20:12:40 INFO ImgRef = /home/user/tmp/melange-guest-1073993080
2024/08/29 20:12:40 INFO running step "Check out sources from git"
2024/08/29 20:12:40 INFO [git checkout] repo='https://github.com/jpr5/ngrep' dest='.' depth='1' branch='' tag='V1_47' expcommit='351ac491358f89a7860bbbe9963cb4bc7d60000f' recurse='false'
2024/08/29 20:12:40 INFO [git checkout] execute: git config --global --add safe.directory /tmp/tmp.ebTd0s
2024/08/29 20:12:40 INFO [git checkout] execute: git config --global --add safe.directory /home/build
2024/08/29 20:12:40 INFO [git checkout] execute: git clone --quiet --origin=origin --config=user.name=Melange Build --config=user.email=melange-build@cgr.dev --config=advice.detachedHead=false --branch=V1_47 --depth=1 https://github.com/jpr5/ngrep /tmp/tmp.ebTd0s
2024/08/29 20:12:40 INFO [git checkout] execute: cd /tmp/tmp.ebTd0s
2024/08/29 20:12:40 INFO [git checkout] tar -c . | tar -C "/home/build" -x
2024/08/29 20:12:40 INFO [git checkout] execute: cd /home/build
2024/08/29 20:12:40 INFO [git checkout] execute: git config --global --add safe.directory /home/build
2024/08/29 20:12:40 INFO [git checkout] execute: git fetch --quiet origin --depth=1 --no-tags +refs/tags/V1_47:refs/origin/tags/V1_47
2024/08/29 20:12:41 INFO [git checkout] execute: git checkout --quiet origin/tags/V1_47
2024/08/29 20:12:41 INFO [git checkout] tag V1_47 is 351ac491358f89a7860bbbe9963cb4bc7d60000f
2024/08/29 20:12:41 WARN autoreconf: export WARNINGS=
2024/08/29 20:12:41 WARN autoreconf: warning: autoconf input should be named 'configure.ac', not 'configure.in'
2024/08/29 20:12:41 WARN autoreconf: Entering directory '.'
2024/08/29 20:12:41 WARN autoreconf: configure.in: not using Gettext
2024/08/29 20:12:42 WARN autoreconf: running: aclocal --force 
2024/08/29 20:12:42 WARN aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
2024/08/29 20:12:42 WARN autoreconf: configure.in: tracing
2024/08/29 20:12:42 WARN autoreconf: configure.in: not using Libtool
2024/08/29 20:12:42 WARN autoreconf: configure.in: not using Intltool
2024/08/29 20:12:42 WARN autoreconf: configure.in: not using Gtkdoc
2024/08/29 20:12:42 WARN autoreconf: running: /usr/bin/autoconf --force
2024/08/29 20:12:42 WARN configure.in:14: warning: The macro 'AC_CANONICAL_SYSTEM' is obsolete.
2024/08/29 20:12:42 WARN configure.in:14: You should run autoupdate.
2024/08/29 20:12:42 WARN ./lib/autoconf/general.m4:2081: AC_CANONICAL_SYSTEM is expanded from...
2024/08/29 20:12:42 WARN configure.in:14: the top level
2024/08/29 20:12:42 WARN configure.in:16: warning: The macro 'AC_HEADER_STDC' is obsolete.
2024/08/29 20:12:42 WARN configure.in:16: You should run autoupdate.
2024/08/29 20:12:42 WARN ./lib/autoconf/headers.m4:663: AC_HEADER_STDC is expanded from...
2024/08/29 20:12:42 WARN configure.in:16: the top level
2024/08/29 20:12:42 WARN autoreconf: running: /usr/bin/autoheader --force
2024/08/29 20:12:42 WARN autoheader: warning: autoconf input should be named 'configure.ac', not 'configure.in'
2024/08/29 20:12:42 WARN autoreconf: configure.in: not using Automake
2024/08/29 20:12:42 WARN autoreconf: './config.sub' is updated
2024/08/29 20:12:42 WARN autoreconf: './config.guess' is updated
2024/08/29 20:12:42 WARN autoreconf: Leaving directory '.'
2024/08/29 20:12:42 INFO running step "Run autoconf configure script"
2024/08/29 20:12:43 INFO
2024/08/29 20:12:43 INFO Configuring System ...
2024/08/29 20:12:43 INFO
2024/08/29 20:12:43 INFO checking build system type... x86_64-pc-linux-gnu
2024/08/29 20:12:43 INFO checking host system type... x86_64-pc-linux-gnu
2024/08/29 20:12:43 INFO checking target system type... x86_64-pc-linux-gnu
2024/08/29 20:12:43 INFO checking for x86_64-pc-linux-gnu-gcc... x86_64-pc-linux-gnu-gcc
2024/08/29 20:12:43 INFO checking whether the C compiler works... yes
2024/08/29 20:12:43 INFO checking for C compiler default output file name... a.out
2024/08/29 20:12:43 INFO checking for suffix of executables... 
2024/08/29 20:12:43 INFO checking whether we are cross compiling... no
2024/08/29 20:12:43 INFO checking for suffix of object files... o
2024/08/29 20:12:43 INFO checking whether the compiler supports GNU C... yes
2024/08/29 20:12:43 INFO checking whether x86_64-pc-linux-gnu-gcc accepts -g... yes
2024/08/29 20:12:43 INFO checking for x86_64-pc-linux-gnu-gcc option to enable C11 features... none needed
2024/08/29 20:12:43 INFO checking for stdio.h... yes
2024/08/29 20:12:43 INFO checking for stdlib.h... yes
2024/08/29 20:12:43 INFO checking for string.h... yes
2024/08/29 20:12:43 INFO checking for inttypes.h... yes
2024/08/29 20:12:43 INFO checking for stdint.h... yes
2024/08/29 20:12:43 INFO checking for strings.h... yes
2024/08/29 20:12:43 INFO checking for sys/stat.h... yes
2024/08/29 20:12:43 INFO checking for sys/types.h... yes
2024/08/29 20:12:43 INFO checking for unistd.h... yes
2024/08/29 20:12:44 INFO checking for grep that handles long lines and -e... /bin/grep
2024/08/29 20:12:44 INFO checking for egrep... /bin/grep -E
2024/08/29 20:12:44 INFO
2024/08/29 20:12:44 INFO Configuring GNU Regular Expression library ...
2024/08/29 20:12:44 INFO
2024/08/29 20:12:44 INFO creating cache ./config.cache
2024/08/29 20:12:45 INFO checking host system type... x86_64-pc-linux-gnu
2024/08/29 20:12:45 INFO checking target system type... x86_64-pc-linux-gnu
2024/08/29 20:12:45 INFO checking build system type... x86_64-pc-linux-gnu
2024/08/29 20:12:45 INFO checking for gcc... gcc
2024/08/29 20:12:45 INFO checking whether the C compiler (gcc -O2 -Wall -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -march=x86-64-v2 -mtune=broadwell -Wl,--as-needed,-O1,--sort-common -Wl,-z,relro,-z,now,-z,noexecstack) works... no
2024/08/29 20:12:45 WARN configure: error: installation or configuration problem: C compiler cannot create executables.
2024/08/29 20:12:45 INFO
2024/08/29 20:12:45 INFO Configuring Network Grep ...
2024/08/29 20:12:45 INFO
2024/08/29 20:12:45 INFO checking for a complete set of pcap headers... checking how to run the C preprocessor... x86_64-pc-linux-gnu-gcc -E
2024/08/29 20:12:45 INFO checking for egrep -e... (cached) /bin/grep -E
2024/08/29 20:12:45 INFO found /usr/include/pcap
2024/08/29 20:12:45 INFO checking for pcap_open_live in -lpcap... yes
2024/08/29 20:12:45 INFO checking for BPF include path... pcap/bpf.h
2024/08/29 20:12:45 INFO checking for DLT_LINUX_SLL in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO checking for DLT_LOOP in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO checking for DLT_IEEE802_11 in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO checking for DLT_IEEE802_11_RADIO in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO checking for DLT_RAW in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO checking for DLT_PFLOG in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO checking for DLT_IPNET in pcap/bpf.h... yes
2024/08/29 20:12:45 INFO
2024/08/29 20:12:45 INFO CONFIG: IPv6 support enabled
2024/08/29 20:12:45 INFO CONFIG: privilege dropping enabled (using nobody)
2024/08/29 20:12:45 INFO CONFIG: using GNU regex library
2024/08/29 20:12:45 INFO CONFIG: tcpkill feature disabled
2024/08/29 20:12:45 INFO CONFIG: automatically including VLAN frames
2024/08/29 20:12:45 INFO
2024/08/29 20:12:45 INFO configure: creating ./config.status
2024/08/29 20:12:45 INFO config.status: creating Makefile
2024/08/29 20:12:45 INFO config.status: creating config.h
2024/08/29 20:12:45 INFO make  -C regex-0.12 regex.o
2024/08/29 20:12:45 INFO make[1]: Entering directory '/home/build/regex-0.12'
2024/08/29 20:12:45 INFO cc -Iregex-0.12 -I/usr/include/pcap -O2 -Wall -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -march=x86-64-v2 -mtune=broadwell  -DHAVE_CONFIG_H -DLINUX  -D_BSD_SOURCE=1 -D__FAVOR_BSD=1 -O2 -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS  -c -o regex.o regex.c
2024/08/29 20:12:45 WARN In file included from /usr/include/sys/types.h:25,
2024/08/29 20:12:45 WARN                  from regex.c:40:
2024/08/29 20:12:45 WARN /usr/include/features.h:197:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
2024/08/29 20:12:45 WARN   197 | # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
2024/08/29 20:12:45 WARN       |   ^~~~~~~
2024/08/29 20:12:45 WARN regex.c:43:10: fatal error: config.h: No such file or directory
2024/08/29 20:12:45 WARN    43 | #include "config.h"
2024/08/29 20:12:45 WARN       |          ^~~~~~~~~~
2024/08/29 20:12:45 WARN compilation terminated.
2024/08/29 20:12:45 INFO make[1]: Leaving directory '/home/build/regex-0.12'
2024/08/29 20:12:45 WARN make[1]: *** [<builtin>: regex.o] Error 1
2024/08/29 20:12:45 WARN make: *** [Makefile:62: regex-0.12/regex.o] Error 2
2024/08/29 20:12:45 ERRO ERROR: failed to build package. the build environment has been preserved:
2024/08/29 20:12:45 INFO   workspace dir: /home/user/tmp/melange-workspace-2747928236
2024/08/29 20:12:45 INFO   guest dir: /home/user/tmp/melange-guest-2689419024
2024/08/29 20:12:45 ERRO failed to build package: unable to run package ngrep pipeline: unable to run pipeline: exit status 2
make[1]: *** [Makefile:111: packages/x86_64/ngrep-1.47-r1.apk] Error 1
make[1]: Leaving directory '/home/user/src/wolfi-os'
make: *** [Makefile:101: package/ngrep] Error 2

```

--- a/ngrep.yaml
+++ b/ngrep.yaml
@@ -1,7 +1,7 @@
 package:
   name: ngrep
   version: 1.47
-  epoch: 1
+  epoch: 2
   description: A grep-like utility that allows you to search for network packets on an interface
   copyright:
     - license: BSD-3-Clause
@@ -15,6 +15,7 @@ environment:
       - busybox
       - ca-certificates-bundle
       - libpcap-dev
+      - openssf-compiler-options
       - pcre-dev
 
 # transform melange version 1.47 => 1_47
@@ -31,7 +32,9 @@ pipeline:
       tag: V${{vars.mangled-package-version}}
       expected-commit: 351ac491358f89a7860bbbe9963cb4bc7d60000f
 
-  - runs: autoreconf -vif
+  - runs: |
+      autoreconf -vif
+      ( cd regex-0.12 && autoreconf -vif )
 
   - uses: autoconf/configure
     with:
@@ -46,6 +49,18 @@ pipeline:
   - uses: autoconf/make-install
 
   - uses: strip
+
+test:
+  pipeline:
+    - name: "check 'ngrep -V' for expected version"
+      runs: |
+        set +x
+        out=$(ngrep -V) || { echo "FAIL: ngrep -V exited $?"; exit 1; }
+        echo "$out" | grep -q "${{package.version}}" || {
+          echo "FAIL: ngrep -V did not contain ${{package.version}}"
+          exit 1
+        }
+        echo "PASS: ngrep -V contained expected version (${{package.version}})"
 
 update:
   version-separator: _

